PreBuild.lean

```diff
@@ -66,16 +66,30 @@ def Variant.writeToLean (v : Variant) : IO Unit := do
   writeln h pref ["| ", v.ident]
 
 def Enum.writeToLean (e : Enum) (skipIfDefs := true) : IO Unit := do
+  -- prepare the definition of `listAll`, populated when iterating over the variants
+  let listAll := "listAll"
+  let mut listAllDef := s!"\
+/-- The list of all variants. -/
+def {listAll} : List {e.ident} := [\
+  "
   let wln := writeln h pref
   let wlns := writelns h pref
+  -- write the type's doc
   e.doc.writeToLean h pref
+  -- type definition
   wln ["inductive ", e.ident, " where"]
+  -- write each variant (and its doc), populate `listAll`'s definition
   for v in e.variants do
-    if skipIfDefs ∧ v.ifdef.isSome then
-      continue
+    -- ignore variants inside `#ifdef`?
+    if skipIfDefs ∧ v.ifdef.isSome then continue
+    -- write variant as part of the type's definition
     v.writeToLean h (pref ++ "  ")
+    -- write the part of `listAllDef` dealing with this variant
+    listAllDef := s!"{listAllDef}\n  {e.ident}.{v.ident},"
+  -- finalize `listAll`'s definition
+  listAllDef := s!"{listAllDef}\n]"
   wlns [
-    ["deriving Inhabited, Repr, BEq"],
+    ["deriving Inhabited, Repr, BEq, DecidableEq"],
```

Comment:
Let's not add `DecidableEq`. While this is fine, I am hesitant to make any formal guarantee about the cvc5 API for now. Let's use `==` and `!=` instead.